plugins/system/menufilecheck/menufilecheck.php

```diff
@@ -0,0 +1,104 @@
+<?php
+// no direct access
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Table\Table;
+use Joomla\CMS\Plugin\CMSPlugin;
+use Joomla\CMS\Factory;
+use Joomla\CMS\Uri\Uri;
+use Joomla\CMS\Application\SiteApplication;
+
+Table::addIncludePath(__DIR__ . '/tables');
+
+class PlgSystemMenuFileCheck extends CMSPlugin
+{
+    protected $db;
+
+    public function __construct(&$subject, $config)
+    {
+        parent::__construct($subject, $config);
+        $this->db = Factory::getDbo();
+    }
+
+    public function onAfterRoute(): bool {
+        $app = Factory::getApplication();
+
+        // Check that we are on the frontend of the site
+        if ($app->isClient('site'))
+        {
+            // Get the current url query parameters
+            Uri::current(); // It's very strange, but without this line at least Joomla 3 fails to fulfill the task
+            $url = Uri::getInstance();
+            $router = SiteApplication::getRouter(); // get router
+            $urlQueryParams = $router->parse($url); // Get the real joomla query as an array - parse current joomla link
+
+            // Check if the url has an Itemid so we know it could be a menu link
+            if (isset($urlQueryParams['Itemid']))
+            {
+                // extract the Itemid value for later use
+                $Itemid = $urlQueryParams['Itemid'];
+
+                // Unset Itemid because link url returned from JMenu doesn't include it and we need to compare it.
+                unset($urlQueryParams['Itemid']);
+
+                // Use the Itemid to get the menu item
+                $menuItem = Factory::getApplication()->getMenu()->getItem((int) $Itemid);
+
+                // Check if the menu item exists
+                if ($menuItem !== false && isset($menuItem->title))
+                {
+                    $menuTitle = $menuItem->title;
+
+                    // Query the rsfiles_files table to check if any file paths match the menu title
+                    $query = $this->db->getQuery(true)
+                        ->select($this->db->quoteName('filepath'))
+                        ->from($this->db->quoteName('#__rsfiles_files'))
+                        ->where('SUBSTRING_INDEX(' . $this->db->quoteName('filepath') . ', "/", 1) = ' . $this->db->quote($menuTitle));
```

Comment:
This will not work correctly for this file path: 

"Sub 1/Sub 2/Sub 3"

It would return Sub 1 but the correct category from this path is Sub 3

I don't know if this can be done in the query or if you will have to do this in the code but look at the other possibilities in the code database and make sure your code handles them.

Test Document Directory/This is NEW Test document.pdf (the category here is Test Document Directory and your code handles this one okay)

Test Document Directory (Since there's no "/" in the title I don't know what the query will do for this one)


New Sub 1/Alpha/Alpha.pdf (This one Alpha is the category but your code would return New Sub 1)